kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/utils/KafkaUtils.java

```diff
@@ -123,7 +122,6 @@ public static boolean restartBroker(String deployNamespace, String clusterName)
             throw new KubeClusterException.NotFound("Kafka cluster name not found!");
         }
         kubeClient().getClient().pods().inNamespace(deployNamespace).withName(podName).withGracePeriod(0).delete();
-        kubeClient().getClient().pods().inNamespace(deployNamespace).withName(podName).waitUntilCondition(Objects::isNull, 5, TimeUnit.MINUTES);
```

Comment:
Perhaps is null was the wrong check but waiting for something to signal the teardown feels right as otherwise waitForDeployment can pass on the only pod being active, I think.

Reply:
You are right, I have missed the possibility of `delete` method to fail and then the following wait won't be taken into account because it is already running. I'll add another check to be sure the pod is null or at least the status is not running, to wait later to be running again.

Reply:
I think we should expand the wait condition to pass if the pod is either in `Succeeded` (maybe `Failed` should be considered ok too) or is `null` based on https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/ 

Though reading about [forced termination](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-termination-forced) makes me think part of the problem is using `0` grace period. As that deletes the pod from the API server without waiting for cleanup.

Reply:
In the other comment from Keith, we realized that we are checking the `Uid` of the pod as well, so I believe with this two checkings (pod is running and new pod has a different uid) is enough to assure that the pod has been restarted.

Reply:
Yeah I know, but only after 5 mins of waiting .

Ok let's merge this and see if it stops being flaky.

Reply:
> Though reading about [forced termination](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-termination-forced) makes me think part of the problem is using `0` grace period. As that deletes the pod from the API server without waiting for cleanup.

Reading this article for forcing the termination, I read this lines that are what we want:
```
When a force deletion is performed, the API server does not wait for confirmation from the kubelet
 that the Pod has been terminated on the node it was running on. It removes the Pod in the API 
immediately so a new Pod can be created with the same name
```

Reply:
Yeah it's deleting it from the API server without waiting to make sure it gets deleted from the node, it could stay running indefinitely. Though unreported by the api server so maybe not our problem